resources/lite_xl_plugin_api.h

```diff
@@ -17,12 +18,25 @@ UNLESS you're us, and you had to modify this file manually to get it ready for 2
 
 Go figure.
 
+NOTE: If you are compiling multiple source files, and want to include "lite_xl_plugin_api.h"
+in all of them, for any file that is not your entrypoint please use
+
+#define LITE_XL_PLUGIN_NON_ENTRYPOINT
+
+Above your #include, or else you'll get link errors.
+
 **/
 
 
 #include <stdarg.h>
 #include <stdio.h> // for BUFSIZ? this is kinda weird
 
+#ifdef LITE_XL_PLUGIN_NON_ENTRYPOINT
+  #define LITE_XL_PLUGIN_API_FUNC
```

Comment:
Makes more sense if you declare `LITE_XL_PLUGIN_ENTRYPOINT`.

Reply:
I specifically want to avoid doing this because I am pretty sure most plugins will only be one .c file; and it only requires you to start making future declarations as you get more complex, rather than out of the gate.

Reply:
That makes 0 sense. If the plugin has 1 file, then there'll be one declaration. If the plugin has 5, then there'll be four. It does not save you any measurable effort, and makes it even harder to find the entry point (because now you gotta _exclude_ all non-entrypoint).

Reply:
The idea is that it makes it more accessible to people that don't actually read, and just want to get started. Given that I am fairly certain most native plugins will be a single .c file, this makes the implementation for those simpler, rather than more complex; make sense?

I take your point, and am sympathetic to it, as that is generally how these things are done. It would also give you a large linking error if you omitted it entirely, so you would notice pretty quickly. I'm down with changing it if you have your heart set on it.

Reply:
The solution is to RTFM.

I think the fact that it is a de-facto way to do things should already play a big part in making decisions.

Reply:
Yeah; I broadly agree with that.